### ISSUE TITLE
Support for TS 2.0 `paths` option is missing

### ISSUE THREAD

**Issue by alexeagle** [ivogabe/gulp-typescript#331]
TypeScript can resolve modules from paths other than `node_modules` as of 2.0.

```
alexeagle-macbookpro2:tmp alexeagle$ cat tsconfig.json
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "baseUrl": "../Projects/angular",
        "paths": {
          "angular2/*": ["dist/js/cjs/angular2/*"]
        }
    },
    "files": ["repro.ts"]
}
alexeagle-macbookpro2:tmp alexeagle$ cat repro.ts
import {bootstrap} from 'angular2/platform/browser';
bootstrap(null);
alexeagle-macbookpro2:tmp alexeagle$ ./node_modules/.bin/tsc
```
shows that TypeScript can compile this, but
```
alexeagle-macbookpro2:tmp alexeagle$ cat gulpfile.js
var gulp = require('gulp');
gulp.task('build', [], function() {
  var tsc = require('gulp-typescript');
  var project = tsc.createProject('./tsconfig.json');
  return project.src().pipe(tsc(project)).js.pipe(gulp.dest('dist'));
});
alexeagle-macbookpro2:tmp alexeagle$ gulp build
[11:39:17] Using gulpfile ~/tmp/gulpfile.js
[11:39:17] Starting 'build'...
repro.ts(1,25): error TS2307: Cannot find module 'angular2/platform/browser'.
[11:39:18] TypeScript: 1 semantic error
[11:39:18] TypeScript: emit succeeded (with errors)
[11:39:18] Finished 'build' after 977 ms
```
shows that gulp-typescript is not propagating that option.

**Comment by ivogabe** [comment]
You have to use a nightly build for this, so you have to add that to your gulpfile:
```javascript
var project = tsc.createProject('./tsconfig.json', { typescript: require('typescript') });
```
Any incompatibilities between TypeScript and gulp-typescript should now be fixed with 2458d2046af72a1776d6833077c275de3955ad65.

**Comment by ivogabe** [comment]
You have to use a nightly build for this, so you have to add that to your gulpfile:
```javascript
var project = tsc.createProject('./tsconfig.json', { typescript: require('typescript') });
```
Any incompatibilities between TypeScript and gulp-typescript should now be fixed with 2458d2046af72a1776d6833077c275de3955ad65.

**Comment by alexeagle** [comment]
thanks!
